### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by bingosummer** [cloudfoundry/bosh-cli#462]
When using `bosh create-env`, I hit multiple times of the following error. Sometimes, a retry could work.
Is it an issue of `bosh-cli` or `stemcell`? Did anyone hit before?
bosh-cli version I used: `5.1.1`

```
Waiting for instance 'bosh/0' to be running...panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x46cc53]

goroutine 2896 [running]:
io.copyBuffer(0x10070c0, 0xc4202f04b0, 0x0, 0x0, 0xc42098e000, 0x8000, 0x8000, 0xc4202f04b0, 0xc4202f04b0, 0xdd2b00)
	/usr/local/go/src/io/io.go:400 +0x143
io.Copy(0x10070c0, 0xc4202f04b0, 0x0, 0x0, 0x20, 0x10, 0x2)
	/usr/local/go/src/io/io.go:362 +0x5a
net.genericReadFrom(0x1005640, 0xc4205a6158, 0x0, 0x0, 0x0, 0x0, 0x403b00)
	/usr/local/go/src/net/net.go:597 +0x84
net.(*TCPConn).readFrom(0xc4205a6158, 0x0, 0x0, 0xc420625e08, 0x40e38d, 0xd9f200)
	/usr/local/go/src/net/tcpsock_posix.go:51 +0xa5
net.(*TCPConn).ReadFrom(0xc4205a6158, 0x0, 0x0, 0x7fbf23ef3288, 0xc4205a6158, 0xc42001c001)
	/usr/local/go/src/net/tcpsock.go:103 +0x5f
io.copyBuffer(0x1005640, 0xc4205a6158, 0x0, 0x0, 0x0, 0x0, 0x0, 0xea7640, 0xc420088700, 0x1005640)
	/usr/local/go/src/io/io.go:386 +0x31a
io.Copy(0x1005640, 0xc4205a6158, 0x0, 0x0, 0x0, 0xc420625f30, 0xc420625f68)
	/usr/local/go/src/io/io.go:362 +0x5a
github.com/cloudfoundry/bosh-cli/deployment/sshtunnel.(*sshTunnel).Start.func3(0x1015260, 0xc4205a6158, 0x0, 0x0, 0xc42031e1e0, 0xc4200a6840)
	/tmp/build/9674af12/gopath/src/github.com/cloudfoundry/bosh-cli/deployment/sshtunnel/ssh_tunnel.go:85 +0xc3
created by github.com/cloudfoundry/bosh-cli/deployment/sshtunnel.(*sshTunnel).Start
	/tmp/build/9674af12/gopath/src/github.com/cloudfoundry/bosh-cli/deployment/sshtunnel/ssh_tunnel.go:84 +0x314
bosh-cli deploy failed!
```

**Comment by NautiluX** [comment]
The bosh cli debug output of the create-env command:
```
[unlimitedRetryStrategy] 2018/08/10 10:05:13 DEBUG - Making attempt #18
[httpClient] 2018/08/10 10:05:13 DEBUG - Sending POST request to endpoint 'https://<redacted>:6868/agent'
[httpAgentClient] 2018/08/10 10:05:13 DEBUG - get_task response value: "applied"
[httpAgentClient] 2018/08/10 10:05:13 WARN - Unable to parse get_task response value: "applied"
[httpClient] 2018/08/10 10:05:13 DEBUG - Sending POST request to endpoint 'https://<redacted>:6868/agent'
[unlimitedRetryStrategy] 2018/08/10 10:05:13 DEBUG - Making attempt #0
[httpClient] 2018/08/10 10:05:13 DEBUG - Sending POST request to endpoint 'https://<redacted>:6868/agent'
[httpAgentClient] 2018/08/10 10:05:13 DEBUG - get_task response value: map[string]interface {}{"agent_task_id":"7d71f267-9294-4560-7cf1-c8d0576a8559", "state":"running"}
[unlimitedRetryStrategy] 2018/08/10 10:05:14 DEBUG - Making attempt #1
[httpClient] 2018/08/10 10:05:14 DEBUG - Sending POST request to endpoint 'https://<redacted>:6868/agent'
[httpAgentClient] 2018/08/10 10:05:14 DEBUG - get_task response value: map[string]interface {}{"agent_task_id":"7d71f267-9294-4560-7cf1-c8d0576a8559", "state":"running"}
<....>
[unlimitedRetryStrategy] 2018/08/10 10:05:53 DEBUG - Making attempt #40
[httpClient] 2018/08/10 10:05:53 DEBUG - Sending POST request to endpoint 'https://<redacted>:6868/agent'
[httpAgentClient] 2018/08/10 10:05:53 DEBUG - get_task response value: map[string]interface {}{"agent_task_id":"7d71f267-9294-4560-7cf1-c8d0576a8559", "state":"running"}
[sshTunnel] 2018/08/10 10:05:54 DEBUG - Received connection
[sshTunnel] 2018/08/10 10:05:54 DEBUG - Dialing local server
[instance] 2018/08/10 10:05:54 WARN - Received SSH tunnel error: Accepting connection on remote server: EOF
[sshTunnel] 2018/08/10 10:05:54 DEBUG - Received connection
[sshTunnel] 2018/08/10 10:05:54 DEBUG - Dialing local server
[instance] 2018/08/10 10:05:54 WARN - Received SSH tunnel error: Accepting connection on remote server: EOF
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x46cc53]

goroutine 5585 [running]:
io.copyBuffer(0xffe4a0, 0xc4205213c0, 0x0, 0x0, 0xc4209ae000, 0x8000, 0x8000, 0xc4205213c0, 0xc4205213c0, 0xdcc180)
        /usr/local/go/src/io/io.go:400 +0x143
io.Copy(0xffe4a0, 0xc4205213c0, 0x0, 0x0, 0xffb3a0, 0xc420081980, 0xc4200d43a1)
        /usr/local/go/src/io/io.go:362 +0x5a
net.genericReadFrom(0xffca00, 0xc42009c6e0, 0x0, 0x0, 0x0, 0x0, 0xc420465c00)
        /usr/local/go/src/net/net.go:597 +0x84
net.(*TCPConn).readFrom(0xc42009c6e0, 0x0, 0x0, 0xc420315e08, 0x40e38d, 0xd98ce0)
        /usr/local/go/src/net/tcpsock_posix.go:51 +0xa5
net.(*TCPConn).ReadFrom(0xc42009c6e0, 0x0, 0x0, 0x7fcec33dab68, 0xc42009c6e0, 0xc42001c001)
        /usr/local/go/src/net/tcpsock.go:103 +0x5f
io.copyBuffer(0xffca00, 0xc42009c6e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xea02e0, 0x0, 0xffca00)
        /usr/local/go/src/io/io.go:386 +0x31a
io.Copy(0xffca00, 0xc42009c6e0, 0x0, 0x0, 0x0, 0xc420465b88, 0xc4200814f8)
        /usr/local/go/src/io/io.go:362 +0x5a
github.com/cloudfoundry/bosh-cli/deployment/sshtunnel.(*sshTunnel).Start.func3(0x100c600, 0xc42009c6e0, 0x0, 0x0, 0xc4203ac7d0, 0xc4205f6540)
        /tmp/build/9674af12/gopath/src/github.com/cloudfoundry/bosh-cli/deployment/sshtunnel/ssh_tunnel.go:85 +0xc3
created by github.com/cloudfoundry/bosh-cli/deployment/sshtunnel.(*sshTunnel).Start
        /tmp/build/9674af12/gopath/src/github.com/cloudfoundry/bosh-cli/deployment/sshtunnel/ssh_tunnel.go:84 +0x314
```

**Comment by mfine30** [comment]
This should be fixed with [v5.1.2](https://github.com/cloudfoundry/bosh-cli/releases/tag/v5.1.2)
